solutions/projectstaffing/jgraph/core/src/main/scala/com/microsoft/graphdataconnect/skillsfinder/service/UserService.scala

```diff
@@ -165,12 +184,17 @@ class UserService {
   }
 
   def isCurrentUserAnAdmin(httpHeaders: HttpHeaders): Boolean = {
-    if (httpHeaders.toSingleValueMap.containsKey("x-ms-client-principal")) {
-      val clientPrincipalToken = httpHeaders.toSingleValueMap.asScala("x-ms-client-principal")
-      isUserPartOfAdminsGroup(clientPrincipalToken)
+    if (anonymousUser) {
+      true
     } else {
-      false
+      if (httpHeaders.toSingleValueMap.containsKey("x-ms-client-principal")) {
```

Comment:
you can simplify if else by using map and get something like :
httpHeaders.toSingleValueMap.asScala.get("x-ms-client-principal").map(isUserPartOfAdminsGroup).getOr(false)

so you don't need whole if/else block.